File: backend/routes/user_routes.js
const express = require('express');
const router = express.Router();
const User = require('../models/User');
const { depositMoney, updateVipStatus } = require('../services/user_service');

router.get("/:user_id", async (req, res) => {
  const user = await User.findById(req.params.user_id);
  if (!user) {
    return res.status(404).json({ error: "User not found." });
  }
  
  const userData = {
    id: user._id.toString(),
    email: user.email,
    name: user.name,
    role: user.role,
    balance: user.balance,
    totalSpent: user.totalSpent,
    orderCount: user.orderCount,
    isVIP: user.isVIP,
    status: user.status,
    warningCount: user.warningCount,
    netComplaints: user.netComplaints,
    demotionsCount: user.demotionsCount
  };
  return res.status(200).json(userData);
});

router.post("/deposit", async (req, res) => {
  const { user_id, amount } = req.body;
  
  if (!user_id || amount === undefined) {
    return res.status(400).json({ error: "user_id and amount are required." });
  }
  
  const [response, status] = await depositMoney(user_id, amount);
  return res.status(status).json(response);
});

router.put("/update-vip/:user_id", async (req, res) => {
  const user = await User.findById(req.params.user_id);
  if (!user) {
    return res.status(404).json({ error: "User not found." });
  }
  
  const updated = await updateVipStatus(req.params.user_id);
  if (!updated) {
    return res.status(400).json({ message: "User does not qualify for VIP status." });
  }
  
  return res.status(200).json({ message: "User promoted to VIP status." });
});

router.get("/pending", async (req, res) => {
  const pendingUsers = await User.find({ status: "PendingApproval" });
  
  const usersList = pendingUsers.map(user => ({
    _id: user._id.toString(),
    email: user.email,
    name: user.name,
    status: user.status
  }));
  
  return res.status(200).json({ users: usersList });
});

module.exports = router;

File: backend/routes/user_routes.py
from flask import Blueprint, request, jsonify
from models.user import User
from services.user_service import deposit_money, update_vip_status

user_bp = Blueprint('users', __name__)

@user_bp.get("/<user_id>")
def get_user(user_id):
    user = User.objects(id=user_id).first()
    if not user:
        return jsonify({"error": "User not found."}), 404
    
    user_data = {
        "id": str(user.id),
        "email": user.email,
        "name": user.name,
        "role": user.role,

        "balance": user.balance,
        "totalSpent": user.totalSpent,
        "orderCount": user.orderCount,
        "isVIP": user.isVIP,
        "status": user.status,
        "warningCount": user.warningCount,
        "netComplaints": user.netComplaints,
        "demotionsCount": user.demotions
    }
    return jsonify(user_data), 200

@user_bp.post("/deposit")
def deposit():
    data = request.get_json()
    user_id = data.get("user_id")
    amount = data.get("amount")

    if not user_id or amount is None:
        return jsonify({"error": "user_id and amount are required."}), 400
    
    response, status = deposit_money(user_id, amount)
    return jsonify(response), status

@user_bp.put("/update-vip/<user_id>")
def update_vip(user_id):
    user = User.objects(id=user_id).first()
    if not user:
        return jsonify({"error": "User not found."}), 404

    updated = update_vip_status(user_id)
    if not updated:
        return jsonify({"message": "User does not qualify for VIP status."}), 400
    
    return jsonify({"message": "User promoted to VIP status."}), 200